### DISCUSSION TITLE
Offline Codespaces

### DISCUSSION THREAD

**Discussion by mrlubos** [community/community#5147] [Codespaces]
Hey there! I have just tried Codespaces for the first time and my understanding is they have a very limited offline support. Are there any plans to make Codespaces available offline? Thanks!

**Comment by tanmayeekamath** [comment]
Hey @mrlubos , can you please elaborate a bit more on offline support? 

Are you familiar with the [VS Code Remote containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containersl)? It basically lets you run a docker container as development environment within VS Code. The devcontainer.json that applies to Codespaces is also used by this extension to access/create a development container with a well-defined tool and runtime stack in an offline mode (without using cloud backed machines that codespaces provide). The only thing to remember while using this extension is that you will need to install Docker on your local machine.  
Here is a doc to learn more on that: https://code.visualstudio.com/docs/remote/containers

Please reach out if you have any questions 🙂

**Reply by mrlubos** [reply]
Hi Tanmayee, thank you for the reply! By offline support, I imagine making source code changes (e.g. commits) while having no connection to the network. When I eventually come online, I would push my changes to the remote repository. This is how I often work without Codespaces, so I wonder if this workflow will be possible with Codespaces as well. So far, I've seen a lot of "reconnecting" messages that were asking me to reload my window. I also read this [article](https://dev.to/n3wt0n/visual-studio-github-codespaces-questions-answered-5ge7) (Q3) about offline support which led me to believe it does not exist.

The extension you sent looks good but no, I am not familiar with it. Is it essentially like running the Codespaces image on my local machine without the need for cloud machines? Am I understanding correctly that I would only need to download the Docker image in advance and then I can safely go offline? If that's right, does it mean I could reduce our Codespaces cost by using this workflow instead of relying on the cloud machines?

**Reply by isibley765** [reply]
@tanmayeekamath are you aware of any similar tool that's still supported? I have a similar want for using code spaces offline, like on my transit into work on public infrastructure, and then pushing changes once I'm back online.

Your link now sends me to a 404 page, but if it's [just moved here](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers), I think this is just what I have installed already, not an offline-capable extension

**Reply by isibley765** [reply]
And @mrlubos did you ever find an answer to your question?
